lib/age.c

```diff
@@ -162,9 +162,14 @@ void agecheck (/*@null@*/const struct spwd *sp)
 		return;
 	}
 
-	remain = sp->sp_lstchg + sp->sp_max - now;
+	remain = sp->sp_lstchg - now;
+	if (remain < LONG_MAX - sp->sp_max) {
+		remain += sp->sp_max;
+	} else {
+		remain = LONG_MAX;
+	}
+
```

Comment:
Why didn't you use sum() here?

Reply:
sum is static in lib/isexpired.c

Reply:
Why not make it extern?

Reply:
The if-statement was simple enough from my point of view. I've reused sum (now addsl) in the function just because adding three values takes one more check than this simple if-statement.

Reply:
You could have a 3-input function called addsl3(), and a 2-input addsl().

Reply:
Here's a suggestion for that:

```c
$ cat addsl.c 
inline long
addsl(long a, long b)
{
	if (a > 0 && b > LONG_MAX - a) {
		errno = EOVERFLOW;
		return LONG_MAX;
	}
	if (a < 0 && b < LONG_MIN - a) {
		errno = EOVERFLOW;
		return LONG_MIN;
	}
	return a + b;
}


static
long cmpl(const void *p1, const void *p2)
{
	long  *l1 = p1;
	long  *l2 = p2;

	if (*l1 < *l2)
		return -1;
	if (*l1 > *l2)
		return +1;
	return 0;
}


inline long
addsl3(long a, long b, long c)
{
	int   e;
	long  n[3];

	n[0] = a;
	n[1] = b;
	n[2] = c;

	e = errno;
	qsort(n, 3, sizeof(long), cmpl);

	errno = 0;
	sum = addsl(n[0], n[2]);
	if (errno == EOVERFLOW)
		return sum;
	sum = addsl(sum, n[1]);
	if (errno == 0)
		errno = e;
	return sum;
}
```

Reply:
Extended the if-conditions a bit so the builtin functions can be used directly. Tested with the test cases specified in PR.